### DISCUSSION TITLE
Working toward a more sleek looking and customizable Zim: Secondary CSS file, accessible style.conf, and preferences links to open them

### DISCUSSION THREAD

**Discussion by rowsof** [zim-desktop-wiki/zim-desktop-wiki#2412] [Ideas]
Hey there! Thanks so much for Zim. I've loved it for probably 10+? years now. So, looong time listener, first time caller.

**Some Background:** I currently use Zim across multiple devices (a Windows machine and a Mac, even!) and am able to confidently keep it synced through google drive because of its stellar architecture. I have modified the style.conf file, and (briefly) explored other styling methods via CSS, but CSS editing has been a bit difficult for a number of reasons, especially across multiple devices. In my brief exploration of styling via CSS, I seemed to discover that editing the CSS overwrites or clears the existing CSS entirely, resulting in lost formatting and a buggy transparent sidebar/header. What we want is to override individual styling/cascade on top of the base CSS file.

I think this request would benefit most of Zim's users, and help bring Zim forward to compete aesthetically with other more modern note-taking apps like obsidian.

### Suggestion Part 1: Empower Zim users to use GTK CSS more easily
- Include a blank secondary style sheet by default, and call it in code after the default GTK theme so that it can cascade on top of the GTK defaults. _I may be totally misunderstanding how GTK actually handles its implementation of CSS, so pardon my ignorance/assumptions._
- Link to this stylesheet from Zim's Preferences so users can easily access/edit.
- Lots of work? Impractical? Bonus request: Allow the user to apply custom ids and classes to text. Would love to be able to link to ids.

### Suggestion Part 2: Allow the user to set a custom preferences location (per project) for all sorts of data, so it is more accessible/less buried in the file system (it's especially buried on Mac), and so it can be shared across devices.
- A new preferences setting for Custom Preferences Location. If information already exists in this location, load it, if it doesn't, ask the user if they'd like to duplicate the current preferences in the specified location. I would likely save my preferences as a sibling of the project content. e.g.
  - Parent folder
    - My Zim Project
    - My Zim Project Preferences
- The custom preferences location also functions as the root location for all of the following: plugins/plugin configs, the secondary stylesheet, and style.conf. Consider also allowing the user to overwrite the locations for these individually as well? IDK.

Even with just part 1 I think we'd start getting some really nice user submitted styles, I'd certainly submit some!

Thanks for the awesome application, feel free to just discard/close this if it's not aligned with Zim's mission.

**Comment by dehesselle** [comment]
I can't tell if you read the GTK documentation, I'm leaning towards "no", so I'm giving you this link: https://docs.gtk.org/gtk3/css-overview.html

There you can find out all about GTK's CSS support. It also mentions how to cascade/import style sheets. ZIm will not be able to go beyond the capabilities of GTK, so the answer to "really support CSS" is "not going to happen".

**Reply by rowsof** [reply]
Fair confusion, and thank you for the link. I should clarify my part 1 to say something more along the lines of "enable Zim to empower its users to use GTK CSS." I'm not a programmer, though I do work adjacent to programmers, and am not a linux user, so my awareness here is limited.
